shared/src/question-generators/math/modTricks.ts

```diff
@@ -0,0 +1,348 @@
+  },
+  de: {
+    name: "Modulare Arithmetik Tricks",
+    description: "Beantworte Fragen zur modularen Arithmetik.",
+    reductionQuestion: "Reduziere ${{x}}$ modulo ${{n}}$.",
+    inverseQuestion: "Finde das Inverse von ${{a}}$ modulo ${{n}}$.",
+    expQuestion: "Berechne ${{a}}^{{{b}}} \\pmod{ {{n}} }$.",
+    expQuestionSuccessiveSquaring: "Berechne ${{a}}^{{{b}}} \\pmod{ {{n}} }$.",
+    bottomTextSuccessiveSquaring: "Sukzessives Quadrieren könnte die Berechnung vereinfachen.",
+    successiveSquaringExplanation: "Sukzessives Quadrieren kann die Berechnung zu vereinfachen.",
+    expQuestionFermat: "Berechne ${{a}}^{{{b}}} \\pmod{ {{n}} }$.",
+    bottomTextFermat:
+      "Der kleine Satz von Fermat kann verwendet werden um den Exponenten zu reduzieren.",
+    simpleQuestion: "Finde eine ganze Zahl $x$, so dass $x \\equiv {{a}} \\pmod{ {{b}} }$.",
+    feedbackInvalid: "Deine Antwort ist ungültig.",
+    feedbackCorrect: "Richtig!",
+    feedbackIncorrect: "Falsch. Die richtige Antwort ist: ${{correctAnswer}}$.",
+    feedbackIncorrectFermat: "Falsch. Die richtige Antwort ist: ${{correctAnswer}}$.",
+    feedbackIncomplete: "Nicht vollständig oder zu komplex",
+  },
+}
+
+export const ModTricks: QuestionGenerator = {
+  id: "modtricks",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["basic math", "modular arithmetic", "modulus", "mod", "arithmetic"],
+  languages: ["en", "de"],
+  author: "Janette Welker",
+  license: "MIT",
+  expectedParameters: [
+    {
+      type: "string",
+      name: "variant",
+      allowedValues: [
+        "simple",
+        "inverse",
+        "exponentiation",
+        "successiveSquaring",
+        "fermatExponentiation",
+      ],
+    },
+  ],
+
+  generate: (lang, parameters, seed) => {
+    const path = serializeGeneratorCall({
+      generator: ModTricks,
+      lang,
+      parameters,
+      seed,
+    })
+
+    const random = new Random(seed)
+    const variant = parameters.variant as
+      | "simple"
+      | "inverse"
+      | "exponentiation"
+      | "successiveSquaring"
+      | "fermatExponentiation"
+
+    let question
+    let testing
+
+    switch (variant) {
+      case "simple":
+        ;({ question, testing } = generateSimpleQuestion(lang, path, random))
+        break
+      case "inverse":
+        ;({ question, testing } = generateInverseQuestion(lang, path, random))
+        break
+      case "exponentiation":
+        ;({ question, testing } = generateExponentiationQuestion(lang, path, random))
+        break
+      case "successiveSquaring":
+        ;({ question, testing } = generateSuccessiveSquaringQuestion(lang, path, random))
+        break
+      case "fermatExponentiation":
+        ;({ question, testing } = generateFermatExponentiationQuestion(lang, path, random))
+        break
+      default:
+        throw new Error("Unknown variant")
+    }
+
+    return {
+      question,
+      testing,
+    }
```

Comment:
Maybe directly return the result/testing. Like in the other generators. Would be simpler to read